helga_jeopardy.py

```diff
@@ -165,37 +214,82 @@ def retrieve_question(client, channel):
 
     """
 
-    logger.debug('initiating question retrieval')
-
-    try:
-        tb_resp = requests.get('{}random.json'.format(api_endpoint))
-    except RequestException:
-        return "Could not retrieve a question from the jservice API"
-
-    json_resp = tb_resp.json()[0]
-    question_text = json_resp['question']
-    answer = json_resp['answer']
-    category = json_resp['category']['title']
-    value = json_resp['value']
-
-    if DEBUG:
-        logger.debug(u'psst! the answer is: {}'.format(answer))
+    if random is True:
+        logger.debug('initiating question retrieval')
+    
+        try:
+            tb_resp = requests.get('{}random.json'.format(api_endpoint))
+        except RequestException:
+            return "Could not retrieve a question from the jservice API"
```

Comment:
It might be useful to log something about the exception here.

```suggestion
        except RequestException as e:
            logger.warn("Error fetching question from jservice API: %d %s", e.response.status_code, e.response.reason)
            return "Could not retrieve a question from the jservice API"
```